crypto/src/proofs/transparent.rs

```diff
@@ -16,6 +16,24 @@ use crate::{
 pub enum Error {
     #[error("Invalid spend auth randomizer")]
     InvalidSpendAuthRandomizer,
+    #[error("Note commitment mismatch")]
+    NoteCommitmentMismatch,
+    #[error("Transmission key mismatch")]
+    TransmissionKeyMismatch,
+    #[error("Value commitment mismatch")]
+    ValueCommitmentMismatch,
+    #[error("Ephemeral public key mismatch")]
+    EphemeralPublicKeyMismatch,
+    #[error("Must not be an identity")]
+    IdentityUnexpected,
+    #[error("Unexpected depth for merkle path")]
+    MerklePathMismatch,
+    #[error("Merkle root mismatch")]
+    MerkleRootMismatch,
+    #[error("Invalid diversified address")]
+    InvalidDiversifiedAddress,
+    #[error("Bad nullifier")]
+    BadNullifier,
```

Comment:
These are probably useful for debugging, but we could eventually replace this whole error enum with a single unit error type, since a snark verification will just succeed or fail, revealing nothing about the witness data.  (Maybe this replacement happens when we implement snark proofs, just noting it for later).